android/src/main/java/com/swmansion/gesturehandler/core/GestureUtils.kt

```diff
@@ -47,4 +55,101 @@ object GestureUtils {
+    if (orientation < 0) {
+      orientation += 2 * PI
+    }
+
+    // To get the same value that we get on web, we have to perform shift by PI/2.
+    // However, if orientation is greater than 3PI/2 we would get angles greater than 2PI. Therefore we simply subtract 3PI/2 in that case.
+    val azimuthAngle = if (orientation >= 3 * PI / 2) {
+      orientation - 3 * PI / 2
+    } else {
+      orientation + PI / 2
+    }
+
+    val tilts = spherical2tilt(altitudeAngle, azimuthAngle)
+
+    stylusData.tiltX = tilts.first
+    stylusData.tiltY = tilts.second
+    stylusData.altitudeAngle = altitudeAngle
+    stylusData.azimuthAngle = azimuthAngle
+    stylusData.pressure = pressure
+  }
+
+  // Source: https://w3c.github.io/pointerevents/#converting-between-tiltx-tilty-and-altitudeangle-azimuthangle
+  private fun spherical2tilt(altitudeAngle: Double, azimuthAngle: Double): Pair<Double, Double> {
+    val eps = 0.000000001
+    val radToDeg = 180 / PI
+
+    var tiltXrad = 0.0
+    var tiltYrad = 0.0
+
+    if (altitudeAngle < eps) {
+      // the pen is in the X-Y plane
+      if (azimuthAngle < eps || abs(azimuthAngle - 2 * PI) < eps) {
+        // pen is on positive X axis
+        tiltXrad = PI / 2
+      }
+      if (abs(azimuthAngle - PI / 2) < eps) {
+        // pen is on positive Y axis
+        tiltYrad = PI / 2
+      }
+      if (abs(azimuthAngle - PI) < eps) {
+        // pen is on negative X axis
+        tiltXrad = -PI / 2
+      }
+      if (abs(azimuthAngle - (3 * PI) / 2) < eps) {
+        // pen is on negative Y axis
+        tiltYrad = -PI / 2
+      }
+      if (azimuthAngle > eps && abs(azimuthAngle - PI / 2) < eps) {
+        tiltXrad = PI / 2
+        tiltYrad = PI / 2
+      }
+      if (abs(azimuthAngle - PI / 2) > eps && abs(azimuthAngle - PI) < eps) {
+        tiltXrad = -PI / 2
+        tiltYrad = PI / 2
+      }
+      if (abs(azimuthAngle - PI) > eps && abs(azimuthAngle - (3 * PI) / 2) < eps) {
+        tiltXrad = -PI / 2
+        tiltYrad = -PI / 2
+      }
+      if (abs(azimuthAngle - (3 * PI) / 2) > eps && abs(azimuthAngle - 2 * PI) < eps) {
+        tiltXrad = PI / 2
+        tiltYrad = -PI / 2
+      }
+    } else {
+      val tanAlt = tan(altitudeAngle)
+
+      tiltXrad = atan(cos(azimuthAngle) / tanAlt)
+      tiltYrad = atan(sin(azimuthAngle) / tanAlt)
+    }
+
+    val tiltX = round(tiltXrad * radToDeg)
+    val tiltY = round(tiltYrad * radToDeg)
+
+    return Pair(tiltX, tiltY)
+  }
+
+  fun createStylusDataObject(stylusData: StylusData): WritableMap {
+    val stylusDataObject = Arguments.createMap()
+
+    stylusDataObject.putDouble("tiltX", stylusData.tiltX)
+    stylusDataObject.putDouble("tiltY", stylusData.tiltY)
+    stylusDataObject.putDouble("altitudeAngle", stylusData.altitudeAngle)
+    stylusDataObject.putDouble("azimuthAngle", stylusData.azimuthAngle)
+    stylusDataObject.putDouble("pressure", stylusData.pressure)
+
+    return stylusDataObject
+  }
```

Comment:
```suggestion
  fun createStylusDataObject(stylusData: StylusData) = Arguments.createMap().apply {
    putDouble("tiltX", stylusData.tiltX)
    putDouble("tiltY", stylusData.tiltY)
    putDouble("altitudeAngle", stylusData.altitudeAngle)
    putDouble("azimuthAngle", stylusData.azimuthAngle)
    putDouble("pressure", stylusData.pressure)
  }
```

This can also be moved to a method on `StylusData` as `toReadableMap`. I think `WritableMap` can be assigned to `ReadableMap` and it makes it clear that it shouldn't be modified.

Reply:
Done in [1ab4c37](https://github.com/software-mansion/react-native-gesture-handler/pull/3111/commits/1ab4c3742ae0dc0bb7317b52438b8693ac0839dc)